sdk/core/azure-core/src/main/java/com/azure/core/util/UrlBuilder.java

```diff
@@ -156,12 +177,24 @@ public UrlBuilder setQuery(String query) {
         return this;
     }
 
+    /**
+     * Clear the query that will be used to build the final URL.
+     *
+     * @return This UrlBuilder so that multiple setters can be chained together.
+     */
+    public UrlBuilder clearQuery() {
+        if (query != null && !query.isEmpty()) {
+            query.clear();
+        }
+        return this;
+    }
+
     /**
      * Get the query that has been assigned to this UrlBuilder.
      *
      * @return the query that has been assigned to this UrlBuilder.
      */
-    public Map<String, String> getQuery() {
+    public Map<String, List<String>> getQuery() {
```

Comment:
> Unfortunately this is a backwards compatibility breaking change. I wonder if it would be better to introduce a `QueryParameter` construct as `azure-core` already has with [Header](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/util/Header.java) and a new API could be provided which returns `Map<String, QueryParameter>`

@alzimmermsft @srnagar @JonathanGiles 

Hi folks, Daniel here engineering manager for the Maps team. Is this suggested fix something we agree on? We'd like an answer so we can proceed. We have another PR depending on this one and blocking the SDK. Thanks!

Reply:
@alzimmermsft is right in observing that the change proposed here is a breaking change to the API. I have no issue with the proposed workaround that Alan suggests - it is probably better to do that than surface a List<String>, as we could provide a number of convenience APIs off the QueryParameter class in time.

Reply:
Agree with @alzimmermsft  and @JonathanGiles. Since this is a breaking change, we'll have to leave this method as-is and introduce a new API that returns a Map<String, QueryParameter>`. However, we should also make some changes to this method that returns `Map<String, String>` to support multiple comma-separated values for a query param.